modules/core/src/main/java/org/apache/ignite/internal/util/lang/IgniteIntObjectTuple.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.util.lang;
+
+import java.io.Externalizable;
+import java.util.Objects;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Convenience class representing mutable tuple of two values when first is int primitive.
+ */
+public class IgniteIntObjectTuple<V> {
```

Comment:
I think this is redundant class for current ticket. `val1` used as columns count. Columns count almost never will be more than 127. Integer values up to 127 by default is cached by JVM (see `Integer.valueOf`). So, there is no overhead for `IgniteBiTuple<Integer, V>` compared to `IgniteIntObjectTuple<V>`. No GC pressure (since new Integer objects are not created), no memory footprint change (since reference to `Integer` costs the same 4 bytes as `int`).

Reply:
fixed.